### ISSUE TITLE
B28 - First node instantiation placed at wrong location

### ISSUE THREAD

**Issue by DolphinDream** [nortikin/sverchok#2605]
The first time you instantiate a node after launching Blender (or starting a new file) it places the node at the wrong location (offset in xy direction relative to the cursor). The subsequent node instantiations are being placed correctly at the cursor location.

My system: MacBook Pro 15”, retina display, Mojave 10.14.6, Blender 2.80

![SV-B28-NewNodePlacement](https://user-images.githubusercontent.com/2083719/66515741-a6c9de00-eaad-11e9-88df-7f5c026e5586.gif)

**Comment by zeffii** [comment]
never noticed that.

the first node placed does have an additional function btw, it instanciates the debug logger ( @portnov ideas? )

**Comment by zeffii** [comment]
that doesn't happen on here. so yeah.. i never noticed that

**Comment by zeffii** [comment]
@portnov @vicdoval   noticed this yet?

**Comment by vicdoval** [comment]
If you don't click on "New" and start the node-tree by adding a node from the search box the node appears where you had the cursor but the view is panned so the origin of the node tree is in the center of the editor. (note that the grid moves during the operation).
I guess the view reset happens on the new node-tree creation but i'm not sure we can (or should) do anything about it

**Comment by DolphinDream** [comment]
@vicdoval correct. This behavior occurs when the editor origin is not centered with the view,  and there is no node tree created when instantiating the first node. The instantiation of a new node creates a new node tree and this seems to recenter the node editor's origin to the view. I checked and it seems that this behavior occurs in 2.79 as well.

**Comment by DolphinDream** [comment]
Apparently this behavior doesn't happen with other node editors (material, compositor etc) since in these cases you cannot add new nodes unless you create a new node tree (material or compositor node tree). It seems that sverchok (by design or perhaps an oversight) lets you instantiate a new node without having a node tree being created first. Without a node tree created it seems that the sverchock add menu has all the nodes grayed out so you can't create one by selecting them from the this menu, however the search menu does let you instantiate the nodes.

**Comment by zeffii** [comment]
what does it really matter.

**Comment by zeffii** [comment]
i added the ability to instantiate a node when no nodetree is present.

i may be able to find out why there's a location issue., but there might be operator chaining going on, and it's possible that the original cursor position becomes disassociated with the add_node_operator event.

**Comment by zeffii** [comment]
it's a little tucked away.

https://github.com/nortikin/sverchok/blob/210262e8b092edafc7d88c7a5dd47a60aa9fbc50/utils/sv_extra_search.py#L143-L161

**Comment by zeffii** [comment]
i hesitate to do anything about this, because i fail to see how it interferes with normal operation.

**Comment by zeffii** [comment]
i get it .. but  ¯\_(ツ)_/¯

**Comment by DolphinDream** [comment]
A “fix” would make SV more polished, but it’s not a must have since there is a workaround... namely create a node tree first :) or don’t shift your editor origin before you create the first node ;) 

The reason why I noticed this happening is that I saved my default startup file to have the SV node editor shown.. (as I’ve being developing SV nodes) and apparently I must have saved it with the origin not centered with the view. I now re-saved my startup file with the origin centered and SV is great again. Of course, for new users who don't know about this issue they may run into this odd placement behavior at some point.

**Comment by zeffii** [comment]
let's see how many people report the issue @DolphinDream  - then entertain the idea of adding a check in the extra_search poll  
```
if not current tree is a sverchok tree type:
    return False

if sverchok_preferences.allow_automatic_nodetree_creation_from_extra_search:
    return True
else 
    return False

...

```
that way no new user will ever see extra search if their nodetree is not sverchoktype and they haven't explicitely gone into preferences to enable that feature. 

haha :)
